src/transformers/models/aria/modeling_aria.py

```diff
@@ -1037,7 +1037,7 @@ def _update_causal_mask(
             target_length = (
                 attention_mask.shape[-1]
                 if isinstance(attention_mask, torch.Tensor)
-                else past_seen_tokens + sequence_length + 1
+                else past_seen_tokens + sequence_length
```

Comment:
This comment applies to all models: the `+1` not needed, we are discarding its corresponding data in the attn forward pass (e.g. [here](https://github.com/huggingface/transformers/blob/2fa876d2d824123b80ced9d689f75a153731769b/src/transformers/models/llama/modeling_llama.py#L220))

By removing it, we should expect a minimal improvement wrt memory and inference time. We also ensure that the causal mask has exactly the length of the (cached) input length, which is a requirement for the downstream use case this PR wants to enable.